### ISSUE TITLE
At 950 epochs, the best results on the zinc dataset are still more than 1.4, for a total of just 1,000 epochs

### ISSUE THREAD

**Issue by QSNZNJING** [bdy9527/Specformer#1]
The experiment seems to have lost some code files，such as pygdataclass.py ...And what is returned in the SpecformerZINC model is a tuple with a different y size. But the above are not the most important, I do not know what is the problem, the code in zinc data set about 950 rounds, the effect is not improved at all, could you please provide a complete code?

**Comment by bdy9527** [comment]
Hi, there. Sorry that I mis-merged the loss functions of ZINC, PCBA, and HIV.

You can replace the following code
```
y_idx = y == y
loss = loss_fn(logits.to(torch.float32)[y_idx], y.to(torch.float32)[y_idx])
```
with 
```
loss = loss_fn(logits.squeeze()，y)
```
for the ZINC dataset. And you will see the correct results.

Here is part of my training log:

<img width="462" alt="image" src="https://github.com/DSL-Lab/Specformer/assets/16743085/ed22d5ea-354c-4fb6-b13a-0c82417bcfba">
